### PR TITLE
wireless/wapi: authentication parameters should update before key set

### PR DESCRIPTION
**Pull request by anchao**
## Summary

wireless/wapi: authentication parameters should update before key set

set authentication parameters before key set on command mode

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

N/A

## Testing

wapi command on bcm43013

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: wireless/wapi/src/wapi.c [hunk 1]
```diff
@@ -521,15 +521,64 @@ static int wapi_essid_cmd(int sock, int argc, FAR char **argv)
...
+      default:
+        return -1;
+    }
+
+  ret = wpa_driver_wext_set_auth_param(sock, argv[0],
+                                       IW_AUTH_WPA_VERSION,
+                                       auth_wpa);
+  if (ret >= 0)
+    {
+      ret = wpa_driver_wext_set_auth_param(sockfd, ifname,
```
Comment (pkarashchenko) [inline wireless/wapi/src/wapi.c:568]:
```suggestion
      ret = wpa_driver_wext_set_auth_param(sock, argv[0],
```

Comment (pkarashchenko) [reply wireless/wapi/src/wapi.c:568]: I still do not see the changes

**Review by pkarashchenko — APPROVED** [review]

File: wireless/wapi/src/wapi.c (hunk 1 above)
Comment (anchao) [reply wireless/wapi/src/wapi.c]: emm... fixed, thank you!

**Review by xiaoxiang781216 — APPROVED** [review]

File: wireless/wapi/src/wapi.c [hunk 2]
```diff
@@ -521,15 +521,64 @@ static int wapi_essid_cmd(int sock, int argc, FAR char **argv)
...
   alg_flag = (enum wpa_alg_e)wapi_str2ndx(argv[2], g_wapi_alg_flags);
 
-  /* Set the Passphrase */
+  if (argc > 3)
+    {
+      auth_wpa = atoi(argv[3]);
+    }
+  else
+    {
+      auth_wpa = CONFIG_NETINIT_WAPI_AUTHWPA;
```
Comment (pkarashchenko) [inline wireless/wapi/src/wapi.c:538]:
```
src/wapi.c: In function 'wapi_psk_cmd':
Error: src/wapi.c:538:18: error: 'CONFIG_NETINIT_WAPI_AUTHWPA' undeclared (first use in this function); did you mean 'CONFIG_INIT_PRIORITY'?
       auth_wpa = CONFIG_NETINIT_WAPI_AUTHWPA;
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                  CONFIG_INIT_PRIORITY
```

Comment (anchao) [reply wireless/wapi/src/wapi.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**